lib/engine/game/g_1822/step/destination_token.rb

```diff
@@ -50,9 +50,12 @@ def available_hex(entity, hex)
           end
 
           def destination_node_check?(entity)
-            destination_hex = entity.destination_coordinates
-            node_keys = @game.graph.connected_nodes(entity).keys
-            node_keys.select(&:city?).any? { |c| c.hex.id == destination_hex }
+            destination_hex = @game.hex_by_id(entity.destination_coordinates)
+            home_node = entity.tokens.first.city
+            destination_hex.tile.nodes.first&.walk(corporation: entity) do |path, _, _|
```

Comment:
22CA destinations are a particular city in that hex. Keep that in mind.

Reply:
Yep, that's on my list.